libs/designsystem/button/src/button.component.ts

```diff
@@ -88,17 +89,54 @@ export class ButtonComponent implements AfterContentInit {
   @Input() isFloating: boolean = false;
   @Input()
   size: ButtonSize | `${ButtonSize}` = ButtonSize.MD;
+  @Input() showIconOnly: boolean = false;
 
-  @ContentChild(IconComponent) icon: IconComponent;
   @ContentChild(IconComponent, { read: ElementRef })
-  iconElementRef: ElementRef<HTMLElement>;
-  private hasText = false;
+  iconElementRef?: ElementRef<HTMLElement>;
 
-  constructor(private elementRef: ElementRef<HTMLElement>) {}
+  constructor(private elementRef: ElementRef<HTMLElement>, private renderer: Renderer2) {}
+
+  private wrapTextNode() {
+    const ifTextNode = (node?: ChildNode): ChildNode | undefined => {
+      return node?.nodeType === Node.TEXT_NODE ? node : undefined;
+    };
+
+    const iconElement = this.iconElementRef?.nativeElement;
+    const textNode =
+      ifTextNode(iconElement?.previousSibling) || ifTextNode(iconElement?.nextSibling);
+    if (textNode) {
+      const placement = textNode === iconElement?.previousSibling ? 'before' : 'after';
```

Comment:
Yes kinda - because in the scenario where there's no icon, this means we'll add the span using `this.renderer.appendChild(parent, textWrapper);` whereas `before` would try to add it using `this.renderer.insertBefore(parent, textWrapper, this.iconElementRef?.nativeElement);`.
But the entire `wrapTextNode()` won't be called if there's no icon(see guard at top of `ngAfterContentInit`)